sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala

```diff
@@ -593,4 +594,61 @@ class DDLParserSuite extends AnalysisTest {
           Seq(Seq("x"), Seq("y"), Seq("a", "b", "c"))))
     }
   }
+
+  test("insert table: append") {
+    parseCompare("INSERT INTO TABLE testcat.ns1.ns2.tbl TABLE source",
+      table("source").insertInto(table("testcat", "ns1", "ns2", "tbl")))
```

Comment:
I would prefer to see parser tests written using the expected case classes instead of calling factory methods. I don't know what the guarantees of these factory methods are, and using the plan nodes directly avoids relying on default values from those factory methods.